src/js/simon/solverSimon.ts

```diff
@@ -0,0 +1,48 @@
+import { showSolution } from '../utilities';
+
+export function simon_solve(): void {
+
+	let result: string[] = [];
+
+	const flash = ["red","blue","green","yellow"];
+
+	const button = [ //[any vowel colours][no vowels colours]
+		[["blue", "yellow", "green"], ["blue", "red", "yellow"]], //red
+		[["red", "green", "red"], ["yellow", "blue", "green"]], //blue
+		[["yellow", "blue", "yellow"], ["green", "yellow", "blue"]], //green
+		[["green", "red", "blue"], ["red", "green", "red"]] //yellow
+	]
```

Comment:
This could be an object of arrays. E.g.
```suggestion
	const button = {
		Red: { Vowel: ["blue", "yellow", "green"], NoVowel: ["blue", "red", "yellow"]},
		Blue: { Vowel: ["red", "green", "red"], NoVowel: ["yellow", "blue", "green"]},
        Green: { Vowel: ["yellow", "blue", "yellow"], NoVowel: ["green", "yellow", "blue"]},
        Yellow: { Vowel: ["green", "red", "blue"], NoVowel: ["red", "green", "red"]},
	}
```

Having said all that, I haven't put any thought into how it would affect the usage of it (or the colours one).